allofplos/plos_corpus.py

```diff
@@ -385,66 +385,67 @@ def download_amended_articles(directory=None, tempdir=newarticledir, amended_art
     return amended_updated_article_list
 
 
-def get_uncorrected_proofs_list(directory=None):
+def get_uncorrected_proofs(directory=None, list_path=uncorrected_proofs_text_list):
     """
     Loads the uncorrected proofs txt file.
     Failing that, creates new txt file from scratch using directory.
     :param directory: Directory containing the article files
-    :return: list of DOIs of uncorrected proofs from text list
+    :return: set of DOIs of uncorrected proofs from text list
     """
     if directory is None:
         directory = get_corpus_dir()
-        
+
     try:
-        with open(uncorrected_proofs_text_list) as f:
-            uncorrected_proofs_list = f.read().splitlines()
+        with open(list_path) as f:
+            uncorrected_proofs = set(f.read().splitlines())
     except FileNotFoundError:
         print("Creating new text list of uncorrected proofs from scratch.")
         article_files = listdir_nohidden(directory)
-        uncorrected_proofs_list = []
+        uncorrected_proofs = set()
         for article_file in tqdm(article_files, disable=None):
             article = Article.from_filename(article_file)
-            if article.proof == 'uncorrected-proof':
-                uncorrected_proofs_list.append(article.doi)
+            article.directory = directory
```

Comment:
hrm… we should infer the directory from the filename… that may be a separate PR, but we should open an issue about that (cause otherwise this has some weird behaviour).

Reply:
indeed; see #52

Reply:
lol! did not realise there was such an old issue on this!